src/01_getting_started/03_state_of_async_rust.md

```diff
@@ -1,24 +1,91 @@
-0.3 `futures` crate. The `async`/`await` language feature is still new.
-Important extensions like `async fn` syntax in trait methods are still
-unimplemented, and the current compiler error messages can be difficult to
-parse.
-
-That said, Rust is well on its way to having some of the most performant
-and ergonomic support for asynchronous programming around, and if you're not
-afraid of doing some spelunking, enjoy your dive into the world of
-asynchronous programming in Rust!
+Parts of async Rust are supported with the same stability guarantees as
+regular Rust. Other parts are still maturing and will change over time.
+Compared to regular Rust, you can expect:
 
+- Outstanding runtime performance.
+- More frequent interaction with advanced language features, such as lifetimes
+  and pinning.
+- Some compatibility constraints, both between sync and async code, and between
+  different async ecosystems.
+- Higher maintenance burden, due to the ongoing evolution of async ecosystems
+  and language support.
+
+In short, async Rust is more difficult to use and can result in a higher
+maintenance burden, but gives you best-in-class performance in return.
+All areas of async Rust are constantly improving,
+so the impact of these issues will wear off over time.
+
+## Language and library support
+
+While asynchronous programming is supported by Rust itself,
+most async applications depend on functionality provided
+by community crates.
+As such, you need to rely on a mixture of
+language features and library support:
+
+- The most fundamental traits, types and functions, such as the
+  [`Future`](https://doc.rust-lang.org/std/future/trait.Future.html) trait
+  are provided by the standard library.
+- The `async/await` syntax is supported directly by the Rust compiler.
+- Many utility types, macros and functions are provided by the
+  [`futures`](https://docs.rs/futures/) crate. They can be used in any async
+  Rust application.
+- Async runtimes, IO and task spawning are provided by "async ecosystems", such
+  as Tokio and async-std. Most async applications, and some async crates,
+  depend on a specific ecosystem. See
+  ["The Async Ecosystem"](../02_execution/01_chapter.md) section for more
+  details.
+
+Some language features you may be used to from regular Rust are not yet
+available in async Rust. Notably, Rust does not let you declare async
+functions in traits. Instead, you need to use workarounds to achieve the same
+result, which can be more verbose.
+
+## Compiling and debugging
+
+Compilation errors in async Rust conform to the same high standards as
+regular Rust, but since async Rust often depends on more complex language
+features, such as lifetimes and pinning, you may encounter these types of
+errors more frequently.
+
+Runtime errors, or panics, also tend to be more verbose, primarily as a
+result of the state machines that asynchronous functions compile to. Stack
```

Comment:
This level of detail about the compiler may not be necessary. The next sentence is also a bit confusing, why would a stack trace not contain functions from the runtime, and why would this make your life more difficult?

Reply:
Got it, makes sense. I'd prefer to not have the compiler "under the hood" details but I'm ok with it if you are. The rephrased second sentence is more clear now.